source/draw/gpu/GpuCanvas.ooc

```diff
@@ -21,16 +21,15 @@ use ooc-collections
 
 import GpuImage, GpuMap, GpuSurface, GpuContext, structs/LinkedList
 
-GpuCanvas: abstract class {
+GpuCanvas: abstract class extends GpuSurface {
 	_target: GpuImage
-	_size: IntSize2D
-	_context: GpuContext
 	blend := false
-	init: func (=_target, =_context)
+	init: func (=_target, context: GpuContext) {
+		super(this _target size, context, FloatTransform2D createScaling(2.0f / this _target size width, 2.0f / this _target size height))
```

Comment:
The transform transfers coordinates from local Quad (-1 to 1) to image coordinates (-width / 2 to width / 2)